mitodl/open-discussions
# Ab/frontpage carusels

#### Pre-Flight checklist
<img width="1677" alt="Screen Shot 2023-03-26 at 10 46 09 PM" src="https://user-images.githubusercontent.com/1934992/227828436-3b636bad-9131-42fb-8fad-3d862b8082bd.png">
<img width="1674" alt="Screen Shot 2023-03-26 at 10 46 24 PM" src="https://user-images.githubusercontent.com/1934992/227828447-0af0552d-3205-4cf7-9fba-0e2910d0c7fa.png">
<img width="1680" alt="Screen Shot 2023-03-26 at 10 46 41 PM" src="https://user-images.githubusercontent.com/1934992/227828458-99ea3d20-9ed8-4612-a750-4c9e975d936e.png">



#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3897

#### What's this PR do?
This pr adds carusels with
Upcoming courses (with a menu for filtering for selecting All, Certificate, Micromasters or Professional ed)
Popular Learning Resources
New Videos from MIT
Upcoming Professional Education Course (in a separate, highlighted box)

The design is here:
https://www.figma.com/file/zyQTnhYbid1DT5ZaS7MCu6/MIT-Open?node-id=582-1535&t=dnW66AbrLKASsvQd-0

The mobile ui for these changes will be fixed and updated to match the designs in a followup pr

Also currently the api filters for filtering upcoming courses for having certificates, Micromasters and Professional ed don't work so all those carusels just return all upcoming courses

Also the learning resource apis don't currently return departments (and not all learning resources have departments)
so the cards still display the resource type instead of the department like the design.

Also there were a few changes to the design of the top search section since that part was implemented. That will also be updated in a follow up pr

#### How should this be manually tested?
Go to http://localhost:8063/infinite

Check that the desktop version of the page looks nice and matches the designs.
